src/osemgrep/language_server/Session.mli

```diff
@@ -1,8 +1,12 @@
 open Lsp
 open Types
 
-type rule_cache = { mutable rules : Rule.t list; lock : Lwt_mutex.t }
-(** Cache of active rules. Protected by mutex as [cache_rules] below can be called asynchronously,
+type session_cache = {
+  mutable rules : Rule.t list;
```

Comment:
maybe some comment before each field explaining the field instead of this global comment at the end of the record?
That would also give the opportunity to explain what "active" mean, and
where the skipped_fingerprints come from.

Reply:
will do